### ISSUE TITLE
Suggested fixes for the bytes type

### ISSUE THREAD

**Issue by dkfellows** [andreas-kupries/critcl#95]
Context: https://stackoverflow.com/q/55814184/301832

I was looking at the `bytes` type and I noticed two odd things about the `s` field in the `critcl_bytes` structure:

1. Tcl's byte array type really does work with `unsigned char *`, so using `char *` for the type of the `s` field is strange.
2. The only safe way to use a passed in byte array is read-only; you can do otherwise if you know the reference count on the `o` is 1… but that's not the common case and hardly anyone uses that. It's also an advanced usage. Making the `s` field be `const` would mean that normal usage would be safe.

In short, `const unsigned char *s;` instead of `char *s;`… and you won't need to have a cast in the code to assign the byte array to that field either.

**Comment by andreas-kupries** [comment]
Done with commit db00b6cd251cfdfca6c685772941e559fdf9b6a1
